utils_nlp/eval/rouge_ext.py

```diff
@@ -0,0 +1,978 @@
+# This script is adopted from https://github.com/Diego999/py-rouge/blob/master/rouge/rouge.py
+# to compute ROUGE scores for non-English languages.
+
+# Currently, the script supports Hindi.
+# Additional language support can be added by adding language specific
+# 1) sentence splitter (SENTENCE_SPLIT_DICT)
+# 2) word tokenizer (WORD_TOKENIZE_DICT)
+# 3) pattern of characters to remove (REMOVE_CHAR_PATTERN_DICT)
+# 4) stemmer (STEMMER_DICT), this is optional since stemming is not applicable to all languages
+# 5) changes to _split_into_words if the text words are not separated by space
+
+# Major changes made to the original rouge.py include:
+# 1) Don't remove non-English or non-numeric characters
+# 2) Removed the ensure_compatibility argument as we don't need to reproduce the results of
+#    the original perl script that only supports English.
+
+
+import re
+import string
+import itertools
+import collections
+
+from indicnlp.tokenize import sentence_tokenize, indic_tokenize
+from ..language_utils.hi.hindi_stemmer import hi_stem
+
+
+class RougeExt:
```

Comment:
can you write RougeExt in a way to inherit Rouge, so that the functions that you didn't change don't need to repeat themselves here?

Reply:
That's a good point! I can try that. It will make the code look much cleaner. 
However, I'm a little concerned about static methods not needed by this class like load_stemmer and load_wordnet_db getting inherited and causing confusion. Thoughts?

Reply:
I think the purpose here is to really capture the change you made and make the knowledge easier to understand. It might involve import the original Rouge, making simpler changes to it and then write an extensible version to it.

Reply:
Yes. I think it's a good idea to inherit from Rouge and did it. 
Then I had to change some static methods to non-static because of the custom options added. For example, to add a custom word splitter, I add to change _split_into_words to non-static, then _get_word_ngrams_and_length and _get_unigrams, then _compute_ngrams and _compute_ngrams_lcs. So the code isn't as clean as I was expecting. 
Please take a look and let me know if you have any suggestions to further simplify it.

Reply:
Yes. I think it's a good idea to inherit from Rouge and did it.
Then I had to change some static methods to non-static because of the custom options added. For example, to add a custom word splitter, I add to change _split_into_words to non-static, then _get_word_ngrams_and_length and _get_unigrams, then _compute_ngrams and _compute_ngrams_lcs. So the code isn't as clean as I was expecting.
Please take a look and let me know if you have any suggestions to further simplify it.